### PR TITLE
remix

### PR DESCRIPTION
**Pull request by rishi-raj-jain**

### CODE REVIEW

File: blog/2024-04-24-secure-remix.mdx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  <NonAdminUser />
+</details>
+
+## Setting up a Remix Project
+
+:::info
+We will use the Phase Two Remix example code here, but the logic could easily be applied to any existing application.
+:::
+
+- Clone the Phase Two [example repo](https://github.com/p2-inc/examples/).
```
Comment (pnzrr) [inline blog/2024-04-24-secure-remix.mdx:47]: These bullet points need to be numbers.

File: blog/2024-04-24-secure-remix.mdx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+- Open the Remix [folder](https://github.com/p2-inc/examples/tree/main/frameworks/remix) within `/frameworks/remix`.
+- Run `npm install` and then `npm run dev -- --port 3000`. This example leverages [remix-auth](https://github.com/sergiodxa/remix-auth) and [remix-keycloak](https://github.com/marsmars0x01/remix-keycloak) to provide hook and HOC support.
+- Open the `app/services/auth.server.ts` file. This is a server only file. We will be updating a few values from the prior section where we set up our OIDC client. Taking the values from the OIDC Client Config section, set those values in the code. While it is recommended to use Environment variables for the secret, for the purpose of this tutorial, paste in the **Client secret** from the OIDC client creation section for the value of `clientSecret`
+
+   ```tsx
+    const keycloakStrategy = new KeycloakStrategy(
+      {
```
Comment (pnzrr) [inline blog/2024-04-24-secure-remix.mdx:54]: let's just reference the object for the values here first, then how those are passed into the function. In the example, we should pull this object out as well to be independent of the function

File: blog/2024-04-24-secure-remix.mdx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+   The logic using the hook to conditionally determine the Authenticated state, can be used to secure routes, components, and more.
+
+1. Open [localhost:3000](http://localhost:3000). You will see the Phase Two example landing page. You current state should be "Not authenticated". Click **Log In**. This will redirect you to your login page.
+
+   :::info
+   Use the non-admin user created in the previous section to sign in.
+   :::
+
+1. Enter the credentials of the non-admin user you created. Click **Submit**. You will then be redirected to the application. The Phase Two example landing page now loads your "Authenticated" state, displaying your user's email and their Token.
```
Comment (pnzrr) [inline blog/2024-04-24-secure-remix.mdx:78]: looks like user only show email and name

File: blog/2024-04-24-secure-remix.mdx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        realm: "shared-deployment-001",
+        clientID: "reg-example-1",
+        clientSecret: "CLIENT_SECRET", // Paste "Client secret" here. Use Environment variables in prod
+        callbackURL: "http://localhost:3000/auth/keycloak/callback",
+      },
+      async ({ profile }) => {
+        return profile
+      }
+    );
+   ```
```
Comment (pnzrr) [inline blog/2024-04-24-secure-remix.mdx:66]: I'd like a bit more explanation here about how the Authenticator users the keycloakStrategy. So order of this should be 
 - Update Keycloak values
 - Those get provided to KeycloakStrategy
 - That strategy is then provided to the Authenticator.

File: blog/2024-04-24-secure-remix.mdx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+   At this point our entire application will be able to access all information and methods needed to perform authentication. View the `session.server.ts` file for additional information about how the `SessionStorage` is used. The `SessionStorage` enables use of Hooks to derive the authenticated state. View `user.tsx` for exactly how the code is authenticating your user. The sections rendering the "Log in" and "Log out" buttons are conditional areas based on the authenticated context. The buttons invoke server-side APIs provided by `remix-auth`.
```
Comment (pnzrr) [inline blog/2024-04-24-secure-remix.mdx:68]: `SessionStorage` is used to store the Keycloak token.

**Review by pnzrr — CHANGES_REQUESTED** [review]

File: blog/2024-04-24-secure-remix.mdx (hunk 2 above)
Comment (rishi-raj-jain) [reply blog/2024-04-24-secure-remix.mdx:54]: how does this look now?

File: blog/2024-04-24-secure-remix.mdx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+:::info
+We will use the Phase Two Remix example code here, but the logic could easily be applied to any existing application.
+:::
+
+1. Clone the Phase Two [example repo](https://github.com/p2-inc/examples/).
+2. Open the Remix [folder](https://github.com/p2-inc/examples/tree/main/frameworks/remix) within `/frameworks/remix`.
+3. Run `npm install` and then `npm run dev -- --port 3000`. This example leverages [remix-auth](https://github.com/sergiodxa/remix-auth) and [remix-keycloak](https://github.com/marsmars0x01/remix-keycloak) to provide HOC support.
+4. Open the `app/services/keycloak.server.ts` file. This is a server only file. We will be updating a few values from the prior section where we set up our OIDC client. Taking the values from the OIDC Client Config section, set those values in the code. While it is recommended to use Environment variables for the secret, for the purpose of this tutorial, paste in the **Client secret** from the OIDC client creation section for the value of `clientSecret`:
+
+  ```tsx
```
Comment (pnzrr) [inline blog/2024-04-24-secure-remix.mdx:52]: Sorry, I wasn't quite clear

```
const kcConfig = {
 realm, 
 ... // other values
}
```

then in the function

```
export default new KeycloakStrategy(
   kcConfig,
   ({ profile }) => profile
  );
```

**Review by pnzrr — CHANGES_REQUESTED** [review]
just a quick change otherwise looks good

File: blog/2024-04-24-secure-remix.mdx (hunk 6 above)
Comment (pnzrr) [reply blog/2024-04-24-secure-remix.mdx:52]: The example https://github.com/p2-inc/examples/pull/4/files should be updated to match

**Review by pnzrr — APPROVED** [review]

**Pull request merged**